.github/workflows/build-test.yml

```diff
@@ -818,6 +817,69 @@ jobs:
         path: ./benchmarks/binsize/**
         name: benchmark-binsize
 
+  # Data size benchmark: track size of provider/testdata/data/testdata.postcard (total data size).
+
+  datasize:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup output data directory
+      run: |
+        mkdir -p benchmarks/binsize/datasize
+
+    - name: Measure size of selected data package provider/testdata/data/testdata.postcard
+      run: |
+        cargo run --package icu_benchmark_binsize  -- provider/testdata/data/testdata.postcard file | tee benchmarks/binsize/datasize/output.txt
+
+    - name: Store benchmark result (non-merge action only)
+      # Data from anything that is not a merge to mainline goes to a preliminary branch
+      if: github.event_name != 'push' || github.ref != 'refs/heads/main'
+      # Use gregtatum special feature to process ndjson-formatted benchmark data
+      uses: gregtatum/github-action-benchmark@d3f06f738e9612988d575db23fae5ca0008d3d12
+      with:
+        tool: 'ndjson'
+        output-file-path: benchmarks/binsize/datasize/output.txt
```

Comment:
nitpick(non-blocking): Remove the `binsize` path segment from `benchmarks/binsize/datasize/...` here and below.  

It makes it seem like the data size testing is a sub-concept of binary size testing. I think of binary size testing as measuring the size of the compiled code only.